zerver/webhooks/pagerduty/view.py

```diff
@@ -166,34 +210,48 @@ def send_formated_pagerduty(
 def api_pagerduty_webhook(
     request: HttpRequest,
     user_profile: UserProfile,
-    payload: Dict[str, Sequence[Dict[str, Any]]] = REQ(argument_type="body"),
+    payload: Dict[str, Any] = REQ(argument_type="body"),
 ) -> HttpResponse:
-    for message in payload["messages"]:
-        message_type = message.get("type")
+    messages = payload.get("messages")
+
+    if messages is not None:
+        for message in messages:
+            message_type = message.get("type")
+
+            # If the message has no "type" key, then this payload came from a
+            # Pagerduty Webhook V2.
+            if message_type is None:
+                break
 
-        # If the message has no "type" key, then this payload came from a
-        # Pagerduty Webhook V2.
-        if message_type is None:
-            break
+            if message_type not in PAGER_DUTY_EVENT_NAMES:
+                raise UnsupportedWebhookEventType(message_type)
 
-        if message_type not in PAGER_DUTY_EVENT_NAMES:
-            raise UnsupportedWebhookEventType(message_type)
+            format_dict = build_pagerduty_formatdict(message)
+            send_formated_pagerduty(request, user_profile, message_type, format_dict)
 
-        format_dict = build_pagerduty_formatdict(message)
-        send_formated_pagerduty(request, user_profile, message_type, format_dict)
+        for message in messages:
+            event = message.get("event")
 
-    for message in payload["messages"]:
-        event = message.get("event")
+            # If the message has no "event" key, then this payload came from a
+            # Pagerduty Webhook V1.
+            if event is None:
+                break
 
-        # If the message has no "event" key, then this payload came from a
-        # Pagerduty Webhook V1.
-        if event is None:
-            break
+            if event not in PAGER_DUTY_EVENT_NAMES_V2:
+                raise UnsupportedWebhookEventType(event)
```

Comment:
Can you add a test for this line, you can look at #18517 for an example of how this has been done.